### PR TITLE
Support pending migrations error for multiple databases.

### LINKED ISSUE #37524: ActiveRecord::PendingMigrationError is not raised when non-primary database migration is down with multi-db
### Steps to reproduce
ActiveRecord::PendingMigrationError is not raised when non-primary database migrations are down with multi-db.

config database.yml 

```
development:
  backbone:
    <<: *default
    database: backbone_development
  library:
    <<: *default
    database: library_development
    migrations_paths: db/library_migrate
```

run the following
```
bin/rails g scaffold User title 
bin/rails g scaffold Book title --db library
bin/rails db:create db:migrate
bin/rails s
# in another console
bin/rails g migration add_published_at_to_books published_at:datetime --db library
```
access http://localhost:3000/users or http://localhost:3000/books

### Expected behavior
Raise ActiveRecord::PendingMigrationError because add_published_at_to_books migration status is down

```
$ bin/rails db:migrate:status

database: backbone_development
 Status   Migration ID    Migration Name
--------------------------------------------------
   up     20191022014341  Create users

database: library_development

 Status   Migration ID    Migration Name
--------------------------------------------------
   up     20191022023152  Create books
  down    20191022023755  Add published at to books
```

```
$ rails db:abort_if_pending_migrations
You have 1 pending migration:
  20191022023755 AddPublishedAtToBooks
Run `rails db:migrate` to update your database then try again.
```
### Actual behavior
There is no error

### System configuration
Rails 6.0.0
Ruby 2.6.5

### PR DESCRIPTION
**Pull request by tgxworld**
### Summary

Resolves https://github.com/rails/rails/issues/37524

### CODE REVIEW

**Comment by tgxworld** [comment]
@eileencodes I decided to have a stab at the issue and was wondering if the approach looks OK to you? There are a few things I still need to fix like

* Ensuring that running pending migrations actually runs migrations for all the active connection pools
* Figure out why tests are not failing locally for me but breaking on CI

**Comment by eileencodes** [comment]
Thanks for taking a look at this!

Unfortunately I don't think this change can work. The Active Record tests work because they have all the connection pools but in a real application development and test environments don't eager load the app by default. I think it's possible to test it in Railties, and definitely reproducible in a demo application. 

The underlying problem is when an app isn't eager loaded then the connection pools won't exist to check pending migrations against. This is why all the rake tasks establish connections before running the tasks. Connections for other models other than ActiveRecord::Base are established when the model is loaded (ie User.first). Unless we establish a connection in pending migrations or start eager loading the app in dev/test I don't think it's possible to fix this, at least not in the short term.

**Comment by tgxworld** [comment]
Thank you for the quick reply! I understand that in the development env, models are not eager loaded. Therefore, the solutions I adopted here is to check for pending migrations once a connection pool with a new connection specification name is created.

File: activerecord/test/cases/migration/pending_migrations_test.rb [hunk 1]
```diff
@@ -100,6 +115,45 @@ def test_understands_migrations_created_out_of_order
             check_pending.call({})
           end
         end
+
+        def test_with_multiple_database
```
Comment (tgxworld) [inline activerecord/test/cases/migration/pending_migrations_test.rb:119]: @eileencodes I've updated the test case here to better reflect what the code is trying to do. If the connection pool for a database with a pending migration has not been created, the pending migration error is not thrown. Instead, we only check for pending migrations when a new connection pool that hasn't been seen by the middleware is detected during the request.

**Comment by tgxworld** [comment]
PR to support multiple database when running pending migrations is @ https://github.com/rails/rails/pull/39757

**Comment by tgxworld** [comment]
@eileencodes I was wondering if you could have another look at this when you're free. Thank you in advanced!

**Comment by eileencodes** [comment]
I'm going to look at this next week. This one is a bit larger so I want to make sure I spend adequate time reviewing and testing it.

**Comment by tgxworld** [comment]
Superseded by https://github.com/rails/rails/pull/46097

**Pull request closed without merging**